src/internal.c

```diff
@@ -20534,8 +20608,16 @@ int SendAlert(WOLFSSL* ssl, int severity, int type)
 
     /* check for available size */
     outputSz = ALERT_SIZE + MAX_MSG_EXTRA + dtlsExtra;
-    if ((ret = CheckAvailableSize(ssl, outputSz)) != 0)
-        return ret;
+    if ((ret = CheckAvailableSize(ssl, outputSz)) != 0) {
```

Comment:
should we set `ssl->options.buildingMsg = 1` before exit in case of `ret == WANT_WRITE && severity != alert_fatal`?

Reply:
I think that the higher layer should have set that don't you think?

Reply:
What do you think about ignoring the alert if it is non fatal and we get a `WANT_WRITE` here?

Reply:
> I think that the higher layer should have set that don't you think?

good point. I think I'm ok with the actual solution then.

> What do you think about ignoring the alert if it is non-fatal and we get a WANT_WRITE here?

I don't know how much a non-fatal alert is important for the protocol.

Reply:
> I don't know how much a non-fatal alert is important for the protocol.

That's a good point. I'll think about how to best solve this issue.

Reply:
Do you think the pending alert solution will solve the scenario you described?